modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/onboarding/pages/Welcome/index.js

```diff
@@ -1,26 +1,45 @@
-import {useContext} from 'react';
+import { useMutation } from '@apollo/client';
+import { useContext, useEffect } from 'react';
 import BaseButton from '../../../../common/components/BaseButton';
 import {
 	onboardingPageGuard,
 	overviewPageGuard,
 	usePageGuard,
 } from '../../../../common/hooks/usePageGuard';
+import { addAccountFlag } from '../../../../common/services/liferay/graphql/queries';
 import Layout from '../../components/Layout';
-import {AppContext} from '../../context';
-import {actionTypes} from '../../context/reducer';
-import {steps} from '../../utils/constants';
+import { AppContext } from '../../context';
+import { actionTypes } from '../../context/reducer';
+import { steps } from '../../utils/constants';
 import WelcomeSkeleton from './Skeleton';
 
-const Welcome = ({userAccount}) => {
-	const [{assetsPath, project}, dispatch] = useContext(AppContext);
-	const {isLoading} = usePageGuard(
+const Welcome = ({ userAccount }) => {
+	const [{ assetsPath, project }, dispatch] = useContext(AppContext);
+	const { isLoading } = usePageGuard(
 		userAccount,
 		onboardingPageGuard,
 		overviewPageGuard,
 		project.accountKey
 	);
+	const [createAccountFlag, { called, loading }] = useMutation(addAccountFlag);
 
-	if (isLoading) {
+	useEffect(() => {
+		if (!isLoading && !called && !loading) {
```

Comment:
The last validation `&& !loading` is not necessary, since you are verifying the called already, what you think?

Reply:
Done. I remove the called variable.

Reply:
called is necessary to avoid loop :(